### DISCUSSION TITLE
Calendar feed for audits, maintenances, etc

### DISCUSSION THREAD

**Discussion by snipe** [snipe/snipe-it#9537] [Ideas]
I'm currently working on an iCal type feed for things like audits, asset maintenances, etc, but I'm not exactly sure how this could look, and I'd love to get some feedback on this.

For many of these events, we don't specify a time. An audit isn't usually scheduled down to the hour/minute, just the day - so assuming we have different calendars for each of these topics (so people can decide which ones they want to subscribe to in their iCal/Outlook/whatever), do I just stack all of those events into individual all-day events? For folks with a bunch of assets, I could see this becoming unreadable/untenable. 

Kinda curious what you all think, how you would use it, IF you would use it, etc. I don't want to bog this down with options just yet until we see how people will use it, but a few options are:

- One all-day event per audit due, maintenance, etc.
- A bunch of audit dues, maintenances, etc stacked into a a single all day event, splitting them into multiple if they exceed XYZ "events" (I'm assuming there is a MB limit on an individual event in the spec, though I haven't exactly checked.)
- ???

What are your thoughts? Is this even something people would use?

**Comment by baditaflorin** [comment]
For example audit per location, or category. 
I would see it to allow some options to either 
- add a one-day event 
- allow to spread multiple events in some time boxes

Something similar to this idea of time blocking https://help.amazingmarvin.com/en/articles/1950240-time-blocking

**Comment by bveina** [comment]
In my institution we have assets distributed over multiple buildings, but the audit requirements are light. i could envision a monthly feed item that breaks down what items need an audit that month, per room or per building. all my location names are ```<building Abbreviation>-<roomNumber>```. then i could perform the audit of those items as time allowed that month.

month 1:
*User Checks Calendar*
inner monologue: "oh, there are lots of items in building A this month. ill set aside some time this/next week and sweep that whole building.

Month 2:
*User Checks Calendar*
inner monologue: " there are only 4 things that need auditing this month but they are scattered through the facility. ill keep my barcode scanner on me, then when i pass through those areas ill audit them.


however, I'm not sure how the calendar feed would handle items that i fail to audit during a month. ideally they would just appear on the next months calendar event.

**Comment by bveina** [comment]
the other item in the initial post that is intriguing is scheduled maintenance. most of my assets need periodic calibration. being able to schedule the next maintenance cycle for equipment would be very handy. but right now on  ```Version v5.1.5 - build 6029 (master)```  i only see the option to add new mainences. not schedule new ones, am i missing something here?

**Comment by King-GO** [comment]
In our case, for critical equipment we need to record maintenance and calibration activities and till year end evaluation to decide next year functional test for each machine.  
So monthly report and upcoming maintenance event show on calendar is needed. 

Suggestion features added:

- Planning & scheduling calibration work
- Tracking history & maintenance records
- Audit (may edit and sort by department or person)
